sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultVolumeSpec.java

```diff
@@ -86,15 +105,33 @@ public int hashCode() {
 
     @Override
     public String toString() {
-        return String.format("%s, type: '%s', container-path: '%s'",
-                super.toString(),
-                getType(),
-                getContainerPath());
+        return ToStringBuilder.reflectionToString(this);
     }
 
     private static Protos.Value scalarValue(double value) {
         Protos.Value.Builder builder = Protos.Value.newBuilder().setType(Protos.Value.Type.SCALAR);
         builder.getScalarBuilder().setValue(value);
         return builder.build();
     }
+
+    private void validateVolume() {
+        validateResource();
+        ValidationUtils.matchesRegex(this, "containerPath", containerPath, VALID_CONTAINER_PATH_PATTERN);
+
+        if (type == Type.MOUNT) {
+            ValidationUtils.nonEmptyAllowNull(this, "profiles", profiles);
```

Comment:
May make sense to also validate that the profiles are valid profile strings. Are there limitations on valid characters in profile names? could check `matchesRegex()` against that definition.

If there are no character limitations then it should probably at least check for `nonBlank()` for each of the provided profile strings.

Reply:
I did in a couple lines after :)
```
ValidationUtils.matchesRegex(this, "profiles[" + index + "]", profile, VALID_PROFILE_PATTERN);
```